contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -351,38 +360,67 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 	log.V(1).Infof("prepared executor command %q with args '%+v'", ci.GetValue(), ci.Arguments)
 
 	// Create mesos scheduler driver.
-	info := &mesos.ExecutorInfo{
+	execInfo := &mesos.ExecutorInfo{
 		Command: ci,
 		Name:    proto.String(execcfg.DefaultInfoName),
 		Source:  proto.String(execcfg.DefaultInfoSource),
 	}
 
 	// Check for staticPods
+	var staticPodCPUs, staticPodMem float64
 	if s.StaticPodsConfigPath != "" {
-		bs, numberStaticPods, err := archive.ZipDir(s.StaticPodsConfigPath)
+		bs, paths, err := archive.ZipDir(s.StaticPodsConfigPath)
 		if err != nil {
 			return nil, nil, err
 		}
-		info.Data = bs
 
-		// Adjust the resource accounting for the executor.
-		// Currently each podTask accounts the default amount of resources.
-		// TODO(joerg84) adapt to actual resources specified by pods.
-		log.Infof("Detected %d staticPods in Configuration.", numberStaticPods)
+		// try to read pod files and sum resources
+		for _, podPath := range paths {
+			podJson, err := ioutil.ReadFile(podPath)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error reading static pod spec: %v", err)
```

Comment:
not super concerned that we don't have a directory watch implemented yet.
we can add it to the TODO list.

On Thu, Jul 30, 2015 at 10:30 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/service/service.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11230#discussion_r35875979>
> :
>
> >
> > -		// Adjust the resource accounting for the executor.
> > -		// Currently each podTask accounts the default amount of resources.
> > -		// TODO(joerg84) adapt to actual resources specified by pods.
> > -		log.Infof("Detected %d staticPods in Configuration.", numberStaticPods)
> > +		// try to read pod files and sum resources
> > +		for _, podPath := range paths {
> > +			podJson, err := ioutil.ReadFile(podPath)
> > +			if err != nil {
> > +				return nil, nil, fmt.Errorf("error reading static pod spec: %v", err)
>
> In fact we need some more dynamic mechanism here: a watched directory.
> That's what the kubelet does. In this sense I totally agree, the error
> behavior should be different.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11230/files#r35875979>
> .
>